### PR TITLE
Fix ingester error propagation bug introduced by #890

### PR DESCRIPTION
**Pull request by tomwilkie**
`memorySeriesError` was begin reported to the user as a 500 here: https://github.com/weaveworks/cortex/blob/master/pkg/distributor/http_server.go#L42

Signed-off-by: Tom Wilkie <tom.wilkie@gmail.com>

### CODE REVIEW

File: pkg/ingester/ingester.go [hunk 1]
```diff
@@ -332,15 +327,10 @@ func (i *Ingester) append(ctx context.Context, sample *model.Sample) error {
 	}()
 
 	prevNumChunks := len(series.chunkDescs)
-	if err := series.add(model.SamplePair{
+	if err := series.add(ctx, model.SamplePair{
 		Value:     sample.Value,
 		Timestamp: sample.Timestamp,
 	}); err != nil {
-		if mse, ok := err.(*memorySeriesError); ok {
```
Comment (leth) [inline pkg/ingester/ingester.go]: I wonder if `err.(*memorySeriesError)` is the part that is wrong, extraneous `*` maybe? The path below was intended to stop `memorySeriesError` leaking back out.

Comment (tomwilkie) [reply pkg/ingester/ingester.go]: Hmm... it looks correct to me.  The issue cleared itself up after the deployment finished so I wonder if I barking up completely the wrong tree?

Comment (tomwilkie) [reply pkg/ingester/ingester.go]: Yeah I'm completely wrong on the cause of this, will do some more digging.

**Comment by leth** [comment]
I think it's safe to close this PR! Please reopen if that's wrong :)

**Pull request closed without merging**